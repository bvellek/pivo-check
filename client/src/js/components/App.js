import React, { Component } from 'react';
import '../../App.css';
import LandingHeader from './LandingHeader';
import LandingPage from './LandingPage';

class App extends Component {
  render() {
    return (
      <div className="App">
        <LandingHeader />
        <LandingPage />
      </div>
    );
  }
}

export default App;
