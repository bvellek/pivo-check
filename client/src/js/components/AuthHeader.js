import React from 'react';
import { Link } from 'react-router-dom';
import SVGLogo from './SVGLogo';

const AuthHeader = () => (
  <header className="auth-header top-header" role="banner">
    <Link to='/'>
      <h1>
        <span className="visually-hidden">Pivo-Check</span>
        <SVGLogo />
      </h1>
    </Link>
  </header>
);

export default AuthHeader;
