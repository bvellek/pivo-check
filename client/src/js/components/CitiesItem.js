import React from 'react';
import { Link } from 'react-router-dom';
import SVGDelete from './SVGDelete';

const CitiesItem = ({
  cityInfo,
}) => (
  <div>
    <div className="city-item">
      <Link to={`/cities/${cityInfo._id}`}><h2 className="city-list-title">{cityInfo.cityName}</h2></Link>
      <div className="city-info">
        <p className="city-list-brew-count">{cityInfo.completedCount}/{cityInfo.brewTotal}</p>
        <form className="city-list-delete-form" action="index.html" method="post">
          <button type="submit" name="button"><SVGDelete /><span className="visually-hidden">Delete {cityInfo.cityName}</span></button>
        </form>
      </div>
    </div>

    <div className="delete-city-warning">
      <h3>Delete {cityInfo.cityName}?</h3>
      <p>Deleting a location will permanently remove it from your checklist and you will lose your visited brewery&nbsp;record.</p>
      <form className="delete-city-form" action="index.html" method="post">
        <button className="keep-city-btn" type="button" name="button">No, Keep {cityInfo.cityName}</button> {' '}
        <button className="delete-city-btn" type="button" name="button"><SVGDelete />&nbsp;Yes, Delete {cityInfo.cityName}</button>
      </form>
    </div>
  </div>
);

export default CitiesItem;
