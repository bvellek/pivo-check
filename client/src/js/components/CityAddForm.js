import React, { Component } from 'react';
import PlacesAutocomplete, { geocodeByAddress } from 'react-places-autocomplete';

class CityAddForm extends Component {
  constructor(props) {
    super(props);
    this.state = { address: '' };
    this.onChange = (address) => this.setState({ address });
  }

  handleFormSubmit = (event) => {
    event.preventDefault();
    const { address } = this.state;

    geocodeByAddress(address, (err, { lat, lng }, results) => {
      if (err) { console.log('Oh no!', err); }

      console.log(`Yay! got latitude and longitude for ${address}`, { lat, lng }, 'whole results: ', results);
      console.log('new result: ', results[0].address_components[0].short_name, ', ', results[0].address_components[2].short_name);
    });
  }

  handleSelect = (address) => {
    this.setState({ address });

    geocodeByAddress(address, (err, { lat, lng }, results) => {
      if (err) { console.log('Oh no!', err); }

      console.log(`Yay! got latitude and longitude for ${address.locality}`, { lat, lng }, 'whole results: ', results);
      console.log('new result: ', results[0].address_components[0].short_name, ', ', results[0].address_components[2].short_name);
    });
  }

  render() {
    const options = {
      componentRestrictions: { country: 'US' },
      types: ['(cities)'],
    };

    const myStyles = {
      root: {
        position: 'relative',
        margin: '0 auto',
        width: '100%',
        maxWidth: '22em',
        padding: '0',
      },
      input: {
        boxSizing: 'border-box',
        width: '100%',
        lineHeight: '2em',
        margin: '0 auto',
        padding: '0 0.2em',
        color: '#44484D',
      },
      autocompleteItemActive: {
        backgroundColor: '#FFBAB3',
      },
    };

    const AutocompleteItem = ({ formattedSuggestion }) => (
      <div>
        <strong>{ formattedSuggestion.mainText }</strong>{' '}
        <small>{ formattedSuggestion.secondaryText }</small>
      </div>
    );

    return (

      <section className='city-add'>
        <form
          className='city-add-form'
          method='post'
          autoComplete='off'
          onSubmit={this.handleFormSubmit}
        >
          <div className='city-form-content'>
            <label htmlFor='city-search'>Add a city</label>
            <PlacesAutocomplete
              value={this.state.address}
              onChange={this.onChange}
              onSelect={this.handleSelect}
              options={options}
              inputId={'city-search'}
              inputName={'city'}
              styles={myStyles}
              autocompleteItem={AutocompleteItem}
              placeholder={'Seattle, WA'}
            />
            <button aria-label='Add city' type='submit' name='addCity'>Submit City 📍</button>
          </div>
        </form>
      </section>
    );
  }
}

export default CityAddForm;

