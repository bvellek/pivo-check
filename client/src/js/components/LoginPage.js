import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import AuthHeader from './AuthHeader';
// import Loader from './Loader';
import Footer from './Footer';

const LoginPage = ({
  onSubmit,
  onChange,
  demoSubmit,
  user,
  registrationSuccessMessage,
  errors,
  // loadingStatus,
}) => (
  <div>
    <AuthHeader />
    <div className="login-page">
      <main role="main">
        <form className="login-form" method="post" onSubmit={onSubmit}>
          <fieldset>
          {registrationSuccessMessage.message ? <p className="success-message">{registrationSuccessMessage.message}</p> : <div /> }
            <legend><h2>Login</h2></legend>
            <label htmlFor="user-email">Email</label>
            <input
              id="user-email"
              type="email"
              name="email"
              value={user.email}
              onChange={onChange}
              required
            />
            {errors.email ? <p className="error-message" aria-live="assertive">
              <span>{errors.email}</span>
            </p> : <div />}
            <label htmlFor="user-password">Password</label>
            <input
              id="user-password"
              type="password"
              name="password"
              value={user.password}
              onChange={onChange}
              required
            />
            {errors.password ? <p className="error-message" aria-live="assertive">
              <span>{errors.password}</span>
            </p> : <div />}
            <button className="login-btn" type="submit" name="button">Log In üçª</button>
            <div className="login-form-links">
              <Link to='/registration'>Register</Link> {' | '}
              <a href="mailto:resetpassword@PivoCheck.com?subject=Password Reset&body=Hi, I forgot my password. Please reset my password.">Forgot Password?</a>
            </div>
          </fieldset>
        </form>
        <h3>
          <label htmlFor='demo-btn'>Try it out!</label>
        </h3>
        <button
          id='demo-btn'
          className="demo-btn"
          type="button"
          name="demo-btn"
          onClick={demoSubmit}
        >Demo üç∫</button>
      </main>
      <Footer />
    </div>
  </div>
);

LoginPage.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  demoSubmit: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired,
  registrationSuccessMessage: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
  // loadingStatus: PropTypes.bool.isRequired,
};

export default LoginPage;

