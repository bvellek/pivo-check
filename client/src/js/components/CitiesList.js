import React from 'react';
import CitiesItem from './CitiesItem';

const CitiesList = ({
  cities,
}) => (
  <section className="cities-list">
    <ul>
      {cities.map((city, index) => (
        <li key={index}><CitiesItem cityInfo={city} /></li>
      ))}
    </ul>
  </section>
);

export default CitiesList;
