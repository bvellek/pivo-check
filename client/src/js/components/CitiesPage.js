import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import InAppHeader from './InAppHeader';
import CityAddForm from './CityAddForm';
import SVGDelete from './SVGDelete';
import Footer from './Footer';

class CitiesPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  componentDidMount() {
    this.props.dispatch(actions.cleanAuth());
  }

  render() {
    return (
      <div>
        <InAppHeader />
        <div className="cities-page">
          <main role="main">
            <h1>My Cities</h1>
            <section className="city-add">
              <CityAddForm />
            </section>

          {/* if no cities added */}
            <section className="no-cities">
              <h2>Add a City!</h2>
              <p>Looks like you haven't added any cities to your checklist. Submit above to add a city full of breweries!</p>
            </section>

        {/* default */}
            <section className="cities-list">
              <ul>
                <li>
                  <div className="city-item">
                    <a href="#"><h2 className="city-list-title">Seattle, WA</h2></a>
                    <div className="city-info">
                      <p className="city-list-brew-count">7/20</p>
                      <form className="city-list-delete-form" action="index.html" method="post">
                        <button type="submit" name="button"><SVGDelete /><span className="visually-hidden">Delete Seattle</span></button>
                      </form>
                    </div>
                  </div>

                  <div className="delete-city-warning">
                    <h3>Delete Seattle?</h3>
                    <p>Deleting a location will permanently remove it from your checklist and you will lose your visited brewery record.</p>
                    <form className="delete-city-form" action="index.html" method="post">
                      <button className="keep-city-btn" type="button" name="button">No, Keep Seattle</button> {' '}
                      <button className="delete-city-btn" type="button" name="button"><SVGDelete />&nbsp;Yes, Delete Seattle</button>
                    </form>
                  </div>
                </li>

                <li>
                  <div className="city-item">
                    <a href="#"><h2 className="city-list-title">Portland, OR</h2></a>
                    <div className="city-info">
                      <p className="city-list-brew-count">2/68</p>
                      <form className="city-list-delete-form" action="index.html" method="post">
                        <button type="button" name="button"><SVGDelete /><span className="visually-hidden">Delete Portland</span></button>
                      </form>
                    </div>
                  </div>
                </li>
                <li>
                  <div className="city-item">
                    <a href="#"><h2 className="city-list-title">FL. Pops Hammock Seminole Village, FL</h2></a>
                    <div className="city-info">
                      <p className="city-list-brew-count">5/19</p>
                      <form className="city-list-delete-form" action="index.html" method="post">
                        <button type="button" name="button"><SVGDelete /><span className="visually-hidden">Delete Austin</span></button>
                      </form>
                    </div>
                  </div>
                </li>
              </ul>
            </section>
          </main>
          <Footer />
        </div>
      </div>
    );
  }
}

export default connect()(CitiesPage);

