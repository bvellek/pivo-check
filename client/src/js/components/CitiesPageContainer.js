import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import CitiesPage from './CitiesPage';
import CitiesList from './CitiesList';
import NoCities from './NoCities';

class CitiesPageContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
    };
  }

  componentDidMount() {
    document.querySelector('head > title').innerHTML = 'My Cities | PIVO-CHECK';
    this.props.dispatch(actions.cleanAuth());
    this.props.dispatch(actions.getCities());
  }

  addCity = (e) => {
    e.preventDefault();
  }

  // changeUser = (e) => {
  //   const field = e.target.name;
  //   const user = this.state.user;
  //   user[field] = e.target.value;

  //   this.setState({
  //     user,
  //   });
  // }

  // processForm = (e) => {
  //   e.preventDefault();
  //   this.props.dispatch(actions.loginUser(this.state.user));
  //   console.log(`email: ${this.state.user.email}`);
  //   console.log(`password: ${this.state.user.password}`);
  // }

  // demoSubmit = (e) => {
  //   e.preventDefault();
  //   this.setState({
  //     user: {
  //       password: 'demoPassword',
  //       email: 'demouser@PivoCheck.com',
  //     },
  //   }, () => {
  //     document.querySelector('.login-btn').click();
  //   });
  //   console.log(`email: ${this.state.user.email}`);
  //   console.log(`password: ${this.state.user.password}`);
  // }

  render() {
    const citiesArr = this.props.myCities;
    let citiesDisplay;
    if (citiesArr.length > 0) {
      citiesDisplay = <CitiesList cities={citiesArr} />;
    } else {
      citiesDisplay = <NoCities />;
    }

    return (
      <CitiesPage
        loadingStatus={this.props.loadingStatus}
        addCity={this.addCity}
        myCities={citiesDisplay}
      />
    );
  }
}

const mapStateToProps = (state) => ({
  loadingStatus: state.cities.citiesLoadingStatus,
  myCities: state.cities.myCities,
});

export default connect(mapStateToProps)(CitiesPageContainer);


// <Route exact path="/login" render={() => ( // eslint-disable-line
//           !this.props.loginRedirect
//           ? (<LoginPage
//             onSubmit={this.processForm}
//             onChange={this.changeUser}
//             demoSubmit={this.demoSubmit}
//             user={this.state.user}
//             registrationSuccessMessage={this.props.registrationSuccessMessage}
//             errors={this.props.errors}
//           />)
//           : <Redirect to="/cities" />
//         )}
//       />
