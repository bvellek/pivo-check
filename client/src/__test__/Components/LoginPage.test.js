import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { LoginPage } from '../../js/components/LoginPage';

describe('<LoginPage />', () => {
  const callback = jest.fn();
  const user = {
    email: 'johnDoe@hotmail.com',
    password: 'passwordYO',
  };
  const testComponent = (<LoginPage
    onSubmit={callback}
    onChange={callback}
    demoSubmit={callback}
    user={user}
    registrationSuccessMessage={{}}
    errors={{}}
    loadingStatus={false}
  />);


  it('renders without crashing', () => {
    shallow(testComponent);
  });
  it('it renders correctly', () => {
    const component = shallow(testComponent);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('should call demoSubmit when demo submit is clicked', () => {
    const wrapper = shallow(testComponent);
    const btn = wrapper.find('.demo-btn');
    btn.simulate('click', {
      preventDefault() {},
    });
    expect(callback).toHaveBeenCalled();
  });
  it('should call onChange when input typed', () => {
    const wrapper = shallow(testComponent);
    const input = wrapper.find('#user-password');
    input.simulate('keyDown', { keyCode: 40 });
    expect(callback).toHaveBeenCalled();
  });
  it('should call onChange when input typed', () => {
    const wrapper = shallow(testComponent);
    const input = wrapper.find('#user-email');
    input.simulate('keyDown', { keyCode: 40 });
    expect(callback).toHaveBeenCalled();
  });
  it('should call onSubmit form submitted', () => {
    const wrapper = shallow(testComponent);
    const form = wrapper.find('.login-form');
    form.simulate('submit', {
      preventDefault: () => {},
    });
    expect(callback).toHaveBeenCalled();
  });
  // it('should render Loader when loadingStatus is true', () => {
  //   const wrapper = shallow(<LoginPage
  //     onSubmit={callback}
  //     onChange={callback}
  //     demoSubmit={callback}
  //     user={user}
  //     registrationSuccessMessage={{}}
  //     errors={{}}
  //     loadingStatus='true'
  //   />);
  //   // console.log(wrapper);
  //   // const form = wrapper.find('.login-form');
  //   // form.simulate('submit', {
  //   //   preventDefault: () => {},
  //   // });
  //   // expect(callback).toHaveBeenCalled();
  // });
});
