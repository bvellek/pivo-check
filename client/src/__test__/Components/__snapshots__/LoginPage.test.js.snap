// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> it renders correctly 1`] = `
<div>
  <AuthHeader />
  <div
    className="login-page"
  >
    <main
      role="main"
    >
      <form
        className="login-form"
        method="post"
        onSubmit={[Function]}
      >
        <fieldset>
          <div />
          <legend>
            <h2>
              Login
            </h2>
          </legend>
          <label
            htmlFor="user-email"
          >
            Email
          </label>
          <input
            id="user-email"
            name="email"
            onChange={[Function]}
            required={true}
            type="email"
            value="johnDoe@hotmail.com"
          />
          <div />
          <label
            htmlFor="user-password"
          >
            Password
          </label>
          <input
            id="user-password"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
            value="passwordYO"
          />
          <div />
          <button
            className="login-btn"
            name="button"
            type="submit"
          >
            Log In üçª
          </button>
          <div />
          <div
            className="login-form-links"
          >
            <Link
              replace={false}
              to="/registration"
            >
              Register
            </Link>
             
             | 
            <a
              href="mailto:resetpassword@PivoCheck.com?subject=Password Reset&body=Hi, I forgot my password. Please reset my password."
            >
              Forgot Password?
            </a>
          </div>
        </fieldset>
      </form>
      <h3>
        <label
          htmlFor="demo-btn"
        >
          Try it out!
        </label>
      </h3>
      <button
        className="demo-btn"
        id="demo-btn"
        name="demo-btn"
        onClick={[Function]}
        type="button"
      >
        Demo üç∫
      </button>
    </main>
    <Footer />
  </div>
</div>
`;
